src/QMCHamiltonians/LatticeDeviationEstimator.cpp

```diff
@@ -0,0 +1,238 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Yubo Yang, paul.young.0414@gmail.com, University of Illinois at Urbana-Champaign
+//
+// File created by: Yubo Yang, paul.young.0414@gmail.com, University of Illinois at Urbana-Champaign
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "QMCHamiltonians/LatticeDeviationEstimator.h"
+#include <OhmmsData/AttributeSet.h>
+
+namespace qmcplusplus
+{
+
+LatticeDeviationEstimator::LatticeDeviationEstimator(ParticleSet& P,ParticleSet& sP, std::string tgroup_in, std::string sgroup_in): 
+  tspecies(P.getSpeciesSet()), sspecies(sP.getSpeciesSet()), spset(sP), tpset(P)
+{ 
+  // get the distance table from quantum particle set
+  // !!!! YY: use addTable instead of getTable b/c cloned ParticleSet may not have initialized distance table with source particle set ( this is true even for the master thread, why? )
+  int tid   = P.addTable(sP); // getTable(sP) does not work with threads
+
+  d_table   = P.DistTables[tid];
+  tgroup    = tgroup_in;
```

Comment:
These (tgroup, sgroup, hdf5_out, per_xyz) can be added to the constructor initializer list 
tgroup(tgroup_in), ... etc